Source/Core/VideoBackends/Metal/MTLUtil.mm

```diff
@@ -0,0 +1,470 @@
+                           toBuffer:buf
+                  destinationOffset:0
+             destinationBytesPerRow:4
+           destinationBytesPerImage:4];
+  [download_encoder endEncoding];
+  [cmdbuf commit];
+  [cmdbuf waitUntilCompleted];
+  u32 outpx;
+  memcpy(&outpx, [buf contents], 4);
+  // Proper fbfetch will double contents, Haswell will return black, and Broadwell will do nothing
+  if (outpx == 0x22446688)
+    return true;  // Skylake+
+  else if (outpx == 0x11223344)
+    return false;  // Broadwell
+  else
+    return false;  // Haswell
+}
+
+void Metal::Util::PopulateBackendInfoFeatures(VideoConfig* config, id<MTLDevice> device)
+{
+  // Initialize DriverDetails first so we can use it later
+  DriverDetails::Vendor vendor = DriverDetails::VENDOR_UNKNOWN;
+  if ([[device name] containsString:@"NVIDIA"])
+    vendor = DriverDetails::VENDOR_NVIDIA;
+  else if ([[device name] containsString:@"AMD"])
+    vendor = DriverDetails::VENDOR_ATI;
+  else if ([[device name] containsString:@"Intel"])
+    vendor = DriverDetails::VENDOR_INTEL;
+  else if ([[device name] containsString:@"Apple"])
+    vendor = DriverDetails::VENDOR_APPLE;
+  const NSOperatingSystemVersion cocoa_ver = [[NSProcessInfo processInfo] operatingSystemVersion];
+  double version = cocoa_ver.majorVersion * 100 + cocoa_ver.minorVersion;
+  DriverDetails::Init(DriverDetails::API_METAL, vendor, DriverDetails::DRIVER_APPLE, version,
+                      DriverDetails::Family::UNKNOWN);
+
+#if TARGET_OS_OSX
+  config->backend_info.bSupportsDepthClamp = true;
+  config->backend_info.bSupportsST3CTextures = true;
+  config->backend_info.bSupportsBPTCTextures = true;
+#else
+  bool supports_mac1 = false;
+  bool supports_apple4 = false;
+  if (@available(iOS 13, *))
+  {
+    supports_mac1 = [device supportsFamily:MTLGPUFamilyMac1];
+    supports_apple4 = [device supportsFamily:MTLGPUFamilyApple4];
+  }
+  else
+  {
+    supports_apple4 = [device supportsFeatureSet:MTLFeatureSet_iOS_GPUFamily4_v1];
+  }
+  config->backend_info.bSupportsDepthClamp = supports_mac1 || supports_apple4;
+  config->backend_info.bSupportsST3CTextures = supports_mac1;
+  config->backend_info.bSupportsBPTCTextures = supports_mac1;
+  config->backend_info.bSupportsFramebufferFetch = true;
+#endif
+
+  config->backend_info.AAModes.clear();
+  for (u32 i = 1; i <= 64; i <<= 1)
+  {
+    if ([device supportsTextureSampleCount:i])
+      config->backend_info.AAModes.push_back(i);
+  }
+
+  // The unified memory path (using shared buffers for everything) performs noticeably better with
+  // bbox even on discrete GPUs (20fps vs 15fps in Super Paper Mario elevator), so default to that.
+  // The separate buffer + manual upload path is left available for testing and comparison.
+  if (char* env = getenv("MTL_UNIFIED_MEMORY"))
+    g_features.unified_memory = env[0] == '1' || env[0] == 'y' || env[0] == 'Y';
+  else
+    g_features.unified_memory = true;
```

Comment:
Dolphin generally doesn't use `getenv` for settings like this. I think it's probably fine, but I think it's worth discussing at least (especially since it seems like this environment variable is checked in 3 places). It might be better to use a compile-time constant, or Dolphin's config system.

Reply:
Has there been any backend-specific settings stored with the config system in the past? I'm curious if there's an "established way" to implement such a thing (or if it should just be gone about with the normal way maybe with an `#ifdef __APPLE__` out for other platforms).

Reply:
Let me just remove it, it's got a lot of code associated with it and it's not helping much for the average case